src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Barry {
+
+    // A class for elements in the taskList, with a name and an indicator of whether the task is marked
+    private abstract static class Task{
```

Comment:
Nice work! However, some suggestions regarding the coding standards and my opinions -